From 9b07271835af176f2f10c0c50b520701e0bfc6e4 Mon Sep 17 00:00:00 2001
From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>
Date: Tue, 10 Oct 2017 10:36:01 +0200
Subject: [PATCH 09/69] Create x86 7.5.0 machine types

RH-Author: Dr. David Alan Gilbert <dgilbert@redhat.com>
Message-id: <20171010103601.7723-2-dgilbert@redhat.com>
Patchwork-id: 77054
O-Subject: [RHV7.5 qemu-kvm-rhev PATCH 1/1] Create x86 7.5.0 machine types
Bugzilla: 1499011
RH-Acked-by: Laurent Vivier <lvivier@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>

Not that much change from 2.9.0

Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/i386/pc_piix.c    | 21 ++++++++++++++++++++-
 hw/i386/pc_q35.c     | 19 ++++++++++++++++++-
 include/hw/i386/pc.h |  9 +++++++++
 3 files changed, 47 insertions(+), 2 deletions(-)

diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index 7f12ce3..935391f 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -1141,6 +1141,21 @@ static void pc_machine_rhel7_options(MachineClass *m)
     m->is_default = 1;
 }
 
+static void pc_init_rhel750(MachineState *machine)
+{
+    pc_init1(machine, TYPE_I440FX_PCI_HOST_BRIDGE, \
+             TYPE_I440FX_PCI_DEVICE);
+}
+
+static void pc_machine_rhel750_options(MachineClass *m)
+{
+    pc_machine_rhel7_options(m);
+    m->desc = "RHEL 7.5.0 PC (i440FX + PIIX, 1996)";
+}
+
+DEFINE_PC_MACHINE(rhel750, "pc-i440fx-rhel7.5.0", pc_init_rhel750,
+                  pc_machine_rhel750_options);
+
 static void pc_init_rhel740(MachineState *machine)
 {
     pc_init1(machine, TYPE_I440FX_PCI_HOST_BRIDGE, \
@@ -1149,8 +1164,12 @@ static void pc_init_rhel740(MachineState *machine)
 
 static void pc_machine_rhel740_options(MachineClass *m)
 {
-    pc_machine_rhel7_options(m);
+    pc_machine_rhel750_options(m);
+    m->alias = NULL;
+    m->is_default = 0;
     m->desc = "RHEL 7.4.0 PC (i440FX + PIIX, 1996)";
+    m->numa_auto_assign_ram = numa_legacy_auto_assign_ram;
+    SET_MACHINE_COMPAT(m, PC_RHEL7_4_COMPAT);
 }
 
 DEFINE_PC_MACHINE(rhel740, "pc-i440fx-rhel7.4.0", pc_init_rhel740,
diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index 575d407..e73097b 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -395,6 +395,20 @@ static void pc_q35_machine_rhel7_options(MachineClass *m)
     SET_MACHINE_COMPAT(m, PC_RHEL_COMPAT);
 }
 
+static void pc_q35_init_rhel750(MachineState *machine)
+{
+    pc_q35_init(machine);
+}
+
+static void pc_q35_machine_rhel750_options(MachineClass *m)
+{
+    pc_q35_machine_rhel7_options(m);
+    m->desc = "RHEL-7.5.0 PC (Q35 + ICH9, 2009)";
+}
+
+DEFINE_PC_MACHINE(q35_rhel750, "pc-q35-rhel7.5.0", pc_q35_init_rhel750,
+                  pc_q35_machine_rhel750_options);
+
 static void pc_q35_init_rhel740(MachineState *machine)
 {
     pc_q35_init(machine);
@@ -402,8 +416,11 @@ static void pc_q35_init_rhel740(MachineState *machine)
 
 static void pc_q35_machine_rhel740_options(MachineClass *m)
 {
-    pc_q35_machine_rhel7_options(m);
+    pc_q35_machine_rhel750_options(m);
+    m->alias = NULL;
     m->desc = "RHEL-7.4.0 PC (Q35 + ICH9, 2009)";
+    m->numa_auto_assign_ram = numa_legacy_auto_assign_ram;
+    SET_MACHINE_COMPAT(m, PC_RHEL7_4_COMPAT);
 }
 
 DEFINE_PC_MACHINE(q35_rhel740, "pc-q35-rhel7.4.0", pc_q35_init_rhel740,
diff --git a/include/hw/i386/pc.h b/include/hw/i386/pc.h
index b808a9b..89bdaa4 100644
--- a/include/hw/i386/pc.h
+++ b/include/hw/i386/pc.h
@@ -1001,6 +1001,15 @@ extern void igd_passthrough_isa_bridge_create(PCIBus *bus, uint16_t gpu_dev_id);
             .value = "on",\
         },
 
+#define PC_RHEL7_4_COMPAT \
+        HW_COMPAT_RHEL7_4 \
+        { /* PC_RHEL7_4_COMPAT from PC_COMPAT_2_9 */ \
+            .driver   = "mch",\
+            .property = "extended-tseg-mbytes",\
+            .value    = stringify(0),\
+        },
+
+
 #define PC_RHEL7_3_COMPAT \
         HW_COMPAT_RHEL7_3 \
         { /* PC_RHEL7_3_COMPAT from PC_COMPAT_2_8 */ \
-- 
1.8.3.1

